Remove hard coded path to zpyi.py

peteches opened the pull request:
So I love this idea, and I'm looking forwrd to taking it through it's paces.
But, I use antigen to manage all my external zsh plugins, themes and scripts. Antigen clones the repo to `~/.antigen/<encoded url of repo>` then sources a given file from inside the repo.

However as there is a hard coded directory in `zpyi.zsh` and as it is not in the expected location the python fallback fails.

I have utilised a smidgin of zsh's crazy variable interpolation stuff to locate the `zpyi.zsh` file and then worked out where the `zpyi.py` script is.

I have also updated the README to include an antigen command to load zpyi.

sakshamsharma commented:
Hi, thanks for this awesome PR! :+1: 
I like this idea of sourcing the `py` from the same dir as the zsh script. Makes it robust!

I used to use antigen, but removed it when finally I realized it's just slowing my zsh down some bits (the only thing I use it prezto with some of it's plugins). I do agree antigen support is nice to have as well.

Mind if you/I update the source repo URL in the README though? :smile: I believe you were testing it, and forgot to change it.

peteches commented:
Yeah sorry about that, stupid late night Codeing!

I've updated the repo url and I've removed the antigen promotion. If you've tested it and decided not to use antigen, it's probably best not to mention my opinions on it's level of awesome-ness!

sakshamsharma commented:
Aah I didn't mean it in that way! I do like it quite a bit, just that I no longer needed it. Doesn't mean people shouldn't use it. I'll add that line back :smiley:

Thanks for this commit! And also, maybe I'll try add installation instructions as a prezto module too. If you use oh-my-zsh, feel free to add instructions for that too.

sakshamsharma commented:
In any case, the reason for hardcoding the directory (for what it's worth) is here:
https://github.com/sakshamsharma/dotfiles-genie/blob/master/conf.yml#L28

I symlink all my dotfiles folders to a common place, where I clone all my repos. This is a config file which is parsed by a neat python script.
